base/strings/faces.jl

```diff
@@ -0,0 +1,779 @@
+# TOML stuff
+
+# """
+#     loadfaces!(facetoml::IO)
+
+# Parse `facetoml` as TOML, and load all faces described.
+# The loading is done with `loadfaces!`, which see.
+
+# Face entries should be of the following form:
+# ```toml
+# [face_name]
+# property = "value"
+# """
+# loadfaces!(facetoml::IO) = loadfaces!(TOML.parse(facetoml))
+
+# """
+#     loadfaces!(facetoml::String)
+
+# Open `facetoml`, parse it as TOML, and load all faces described.
+# The loading is done with `loadfaces!`, which see.
+# """
+# loadfaces!(facetoml::String) =
+#     open(facetoml, "r") do io loadfaces!(io) end
+
+const RGBTuple = NamedTuple{(:r, :g, :b), NTuple{3, UInt8}}
+
+"""
+    struct SimpleColor
+
+A basic representation of a color, intended for string styling purposes.
+It can either contain a named color (like `:red`), or an `RGBTuple` which
+is a NamedTuple specifying an `r`, `g`, `b` color with a bit-depth of 8.
+
+## Constructors
+
+```julia
+SimpleColor(name::Symbol)  # e.g. :red
+SimpleColor(rgb::RGBTuple) # e.g. (r=1, b=2, g=3)
+SimpleColor(r::Integer, b::Integer, b::Integer)
+SimpleColor(rgb::UInt32)   # e.g. 0x123456
+```
+
+Also see `tryparse(SimpleColor, rgb::String)`.
+"""
+struct SimpleColor
+    value::Union{Symbol, RGBTuple}
+end
+
+SimpleColor(r::Integer, g::Integer, b::Integer) = SimpleColor((; r=UInt8(r), g=UInt8(g), b=UInt8(b)))
+SimpleColor(rgb::UInt32) = SimpleColor(reverse(reinterpret(UInt8, [rgb]))[2:end]...)
```

Comment:
Should there be another function for handling 256 colors? https://www.ditig.com/256-colors-cheat-sheet

Maybe `SimpleColor(ansi_value::UInt8) = ...`?

Reply:
I don't think so, I think we only want to support 4-bit colour and 24-bit RGB. We fall back from 24-bit RGB to 8-bit colour automatically when needed.